modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/portlet/action/ImportObjectDefinitionMVCActionCommand.java

```diff
@@ -56,52 +57,52 @@ protected void doProcessAction(
 			ActionRequest actionRequest, ActionResponse actionResponse)
 		throws Exception {
 
-		try {
-			_importObjectDefinition(actionRequest);
-		}
-		catch (Exception exception) {
-			if (_log.isDebugEnabled()) {
-				_log.debug(exception);
-			}
+		JSONObject jsonObject = _importObjectDefinition(actionRequest);
 
+		if (jsonObject != null) {
 			HttpServletResponse httpServletResponse =
 				_portal.getHttpServletResponse(actionResponse);
 
 			httpServletResponse.setStatus(HttpServletResponse.SC_BAD_REQUEST);
 
-			JSONObject jsonObject = null;
+			JSONPortletResponseUtil.writeJSON(
+				actionRequest, actionResponse, jsonObject);
 
-			if (exception instanceof ObjectDefinitionNameException) {
-				Class<?> clazz = exception.getClass();
+			hideDefaultSuccessMessage(actionRequest);
```

Comment:
@G-Sa before your changes we were calling this method outside of catch. Shouldn't we maintain the current behavior?